runner.rb

```diff
@@ -1,69 +1,100 @@
-    alignment = gets.chomp
-  end until p1.place_ship(current_ship, location, alignment)
+      current_ship = @player_ships.shift
 
-end
+      begin
+        puts "Please give a location and alignment to place your #{current_ship.class} (#{current_ship.hullpoints} spaces)"
+        print "Location: "
+        location = symbolify(gets.chomp)
+        print "Alignment(h = horizonal, v = verticle): "
+        alignment = gets.chomp
+      end until @player.place_ship(current_ship, location, alignment)
+    end
 
+  end
 
-bill = Bill.new
-b_carrier = Carrier.new(player = true) #uncomment to see Bill's ships
-b_battleship = Battleship.new(player = true) #uncomment to see Bill's ships
-b_cruiser = Cruiser.new(player = true) #uncomment to see Bill's ships
-b_submarine = Submarine.new(player = true) #uncomment to see Bill's ships
-b_destroyer = Destroyer.new(player = true) #uncomment to see Bill's ships
-bill.place_ships([b_carrier, b_battleship, b_cruiser, b_submarine, b_destroyer])
+  def turns
+    until @player.game_over? || @opponent.game_over?
+
+      puts "Bill's Board"
+      # puts @opponent
+      View.display_board(@opponent.to_s)
+      puts '-' * 10
+      puts "Your Board"
+      # puts @player
+      View.display_board(@player.to_s)
+
+      print "pick a target: "
+      input = gets.chomp
+      @opponent.target(symbolify(input))
+
+
+      winner = "You" if @opponent.game_over?
 
-until p1.game_over? || bill.game_over?
 
-  puts "Bill's Board"
-  puts bill
-  puts '-' * 10
-  puts "Your Board"
-  puts p1
+      puts "Bill is thinking..."
+      sleep 1
 
-  print "pick a target: "
-  input = gets.chomp
-  bill.target(symbolify(input))
+      bills_target = @opponent.give_target
+      target_result = @player.target(bills_target)
+      @opponent.response(bills_target, target_result)
+      # check
+      puts "bill's target: #{bills_target}"
+      puts "bill's target result: #{target_result}"
+      puts "bill's last move: #{@opponent.last_hit}"
+      puts "bill's priority targets: #{@opponent.priority_targets}"
 
+      winner = "Bill" if @player.game_over?
 
-  winner = "You" if bill.game_over?
+    end
 
+    puts "#{winner} won!"
 
-  puts "Bill is thinking..."
-  sleep 1
+  end
 
-  bills_target = bill.give_target
-  target_result = p1.target(bills_target)
-  bill.response(bills_target, target_result)
-  puts "bill's target: #{bills_target}"
-  puts "bill's target result: #{target_result}"
-  puts "bill's last move: #{bill.last_hit}"
-  puts "bill's priority targets: #{bill.priority_targets}"
+  private
+
+  def symbolify(input)
+    input.to_sym
+  end
 
-  winner = "Bill" if p1.game_over?
 
 end
 
-puts "#{winner} won!"
\ No newline at end of file
+
+player_ships = [Carrier.new(player = true), Battleship.new(player = true), Cruiser.new(player = true), Submarine.new(player = true), Destroyer.new(player = true)]
+
+
+bill = Bill.new
+b_carrier = Carrier.new(player = true) #uncomment to see Bill's ships
```

Comment:
Why do U keep doing `player = true`? Just Carrier.new(true). Or using the hash style initialization `Carrier.new(player: true)` could make sense...